docker/docker_jit_monitor/src/main.rs

```diff
@@ -115,47 +119,66 @@ enum SpawnRunnerError {
     NoHdcDeviceFound,
     #[error("Failed to list USB devices")]
     LsUsbError,
+    #[error("Failed to deserialize list runners api")]
+    ListRunnersDeserialize(serde_json::Error),
 }
 
-// todo: add arg for optional device to pass into the runner
-fn spawn_runner(config: &RunnerConfig) -> Result<process::Child, SpawnRunnerError> {
-    // Note: octocrab apparently requires more coarse grained tokens compared to `gh`, so we use `gh`.
+/// Function to call the api. Raw just is used spawnrunner.
+/// This gives you the _executed_ cmd.
+/// Notice that the api_endpoint needs a slash before it. The api is very peculious
+/// with slashes and this is the easiest
+/// Note: octocrab apparently requires more coarse grained tokens compared
+/// to `gh`, so we use `gh`.
+fn call_github_runner_api(
+    ci_scope: &str,
+    method: &str,
+    api_endpoint: &str,
+    raw: Option<&RunnerConfig>,
+) -> Result<Output, SpawnRunnerError> {
     let mut cmd = Command::new("gh");
-    let api_endpoint = format!(
-        "{}/actions/runners/generate-jitconfig",
-        config.servo_ci_scope
-    );
+    let api_endpoint = format!("{ci_scope}/actions/runners{api_endpoint}");
     cmd.args([
         "api",
         "--method",
-        "POST",
+        method,
         "-H",
         "Accept: application/vnd.github+json",
         "-H",
         "X-GitHub-Api-Version: 2022-11-28",
         &api_endpoint,
     ]);
-    for label in &config.labels {
-        cmd.arg("--raw-field").arg(format!("labels[]={label}"));
+    if let Some(config) = raw {
+        for label in &config.labels {
+            cmd.arg("--raw-field").arg(format!("labels[]={label}"));
+        }
+        cmd.arg("--raw-field")
+            // Todo: perhaps have a count here? Or add information if it has a device or not
```

Comment:
I think this todo can be removed now, since we do set a count and suffix as part of the name.